src/components/NavBar/NavBar.spec.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+
+import NavBar from './NavBar.component';
+import DarThemeProvider from '../../providers/DarkMode';
+import AuthProvider from '../../providers/Auth';
+import { storage } from '../../utils/storage';
+
+jest.mock('../../utils/storage');
+
+describe('Testing NavBar.component', () => {
+  test('toggleDarkmode when click toggle button', () => {
+    const { container } = render(
+      <AuthProvider>
+        <DarThemeProvider>
+          <NavBar />
+        </DarThemeProvider>
+      </AuthProvider>
+    );
+
+    const darkModeToggle = container.querySelector('#toggleTheme');
+    fireEvent.click(darkModeToggle);
+
+    expect(storage.set).toHaveBeenCalledTimes(1);
+    fireEvent.click(darkModeToggle);
+    expect(storage.set).toHaveBeenCalledTimes(2);
+  });
+
+  test('Search value should change when input', () => {
+    render(
+      <AuthProvider>
+        <DarThemeProvider>
+          <NavBar />
+        </DarThemeProvider>
+      </AuthProvider>
+    );
+
+    const input = screen.getByPlaceholderText(/Search/i);
+    expect(input.value).toBe('');
+    fireEvent.change(input, { target: { value: '23' } });
```

Comment:
Since this updates the state of the component I suggest you to wrap the `fireEvent` into an `act`. This might not have any impact right now, but as a rule of thumb, anything that mutates the state should be wrapped inside an `act`. This is also exported from `@testing-library/react`.